Lib/pdb.py

```diff
@@ -2435,30 +2435,52 @@ def main():
     parser.add_argument('-c', '--command', action='append', default=[], metavar='command', dest='commands',
                         help='pdb commands to execute as if given in a .pdbrc file')
     parser.add_argument('-m', metavar='module', dest='module')
-    parser.add_argument('args', nargs='*',
-                        help="when -m is not specified, the first arg is the script to debug")
 
     if len(sys.argv) == 1:
         # If no arguments were given (python -m pdb), print the whole help message.
         # Without this check, argparse would only complain about missing required arguments.
         parser.print_help()
         sys.exit(2)
 
-    opts = parser.parse_args()
+    opts, args = parser.parse_known_args()
+
+    if opts.module:
+        # If a module is being debugged, we consider the arguments after "-m module" to
+        # be potential arguments to the module itself. We need to parse the arguments
+        # before "-m" to check if there is any invalid argument.
+        # e.g. "python -m pdb -m foo --spam" means passing "--spam" to "foo"
+        #      "python -m pdb --spam -m foo" means passing "--spam" to "pdb" and is invalid
+        idx = sys.argv.index('-m')
+        args_to_pdb = sys.argv[1:idx]
+        # This will automatically raise an error if there are invalid arguments
+        parser.parse_args(args_to_pdb)
+    else:
+        # If a script is being debugged, then pdb expects a script as the first argument.
+        # Anything before the script is considered an argument to pdb itself, which would
+        # be invalid because it's not parsed by argparse.
+        invalid_args = []
+        for arg in args:
+            if not arg.startswith('-'):
+                break
+            invalid_args.append(arg)
+
+        if invalid_args:
```

Comment:
```suggestion
        if any(itertools.takewhile(lambda a: a.startswith('-'), args)):
```

Reply:
This is a better expression. Just one question - is it okay to import `itertools` in `pdb`? We do already use plenty of internal libraries, but there were concerns about using `pdb` to debug stdlib when they are not working.

Reply:
Also, we need to full list of invalid arguments to report to keep the old behavior I think. So we need to turn this `takewhiel` to a list and check that.

Reply:
pdb imports traceback which imports itertools.

Reply:
I think if the stdlib is broken you probably need to debug it on a previous version of python.

Reply:
Okay I used `takewhile` for invalid arguments now, but we do need the full list so I converted it to a list as well.